### DISCUSSION TITLE
Monitoring Index Creation Process and Related Questions

### DISCUSSION THREAD

**Discussion by antohka151** [milvus-io/milvus#30201] [Q&A and General discussion]
Dear Community,

I am currently using the MilvusClient to manage my Milvus database and have successfully created a new collection with on-disk indexes. However, I am unsure of how to monitor the index creation process. I am interested in knowing the current count of indexes that have been created. As I am adding data in the range of hundreds of millions, I would like to understand how to monitor this process going forward. This will be particularly useful as I anticipate creating even larger collections in the future.

Additionally, I have a few questions regarding the indexing process:

-  Will the index be recreated for newly added data to the collection?
-  During the index creation process, is it possible to still perform searches in the database using previously created indexes?

Thank you in advance for your assistance :pray:

**Comment by yhmo** [comment]
Milvus splits data into segments and builds an independent index for each segment.

There are two major kinds of segments: growing segment and sealed segment
- Growing segment is for newly inserted data, once a growing segment's size reaches a threshold, the growing segment is turned into a sealed segment.
- Sealed segment is constant, the data in this segment cannot be changed. Milvus builds an independent index for each sealed segment.

When a search request is received:
- milvus performs brute-force search on all the growing segments, each segment returns a topk result.
- milvus performs index search on all the indexed sealed segments, each segment returns a topk result.
- If some sealed segments are not indexed, milvus performs brute-force search on them and each segment returns a topk result.
- milvus conbines the N topk results into a final topk result and returns to the client.
